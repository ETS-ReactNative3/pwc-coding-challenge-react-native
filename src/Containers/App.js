import React, { Component } from 'react'
import { Provider } from 'mobx-react'

// Components
import NavigationViews from './NavigationViews'

// Store
import Store from '../store'

// Set store (but don't reset for hot reloading)
if(!window.store){ const store = window.store = new Store() }

// The provider of the mobx store to all components is set
export default App = () => 
    <Provider store={store}>
        <NavigationViews />
    </Provider>