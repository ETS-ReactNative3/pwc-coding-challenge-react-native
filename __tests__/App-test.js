/**
 * @format
 */

import 'react-native';
import 'jest-styled-components/native'

import React from 'react';
import App from '../src/Components/';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(App).toMatchSnapshot()
});
