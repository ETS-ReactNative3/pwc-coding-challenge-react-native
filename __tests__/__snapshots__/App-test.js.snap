// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "attrs": Array [],
  "displayName": "Styled(View)",
  "inlineStyle": InlineStyle {
    "rules": Array [
      "
    padding:8px;
    padding-top:0;
",
    ],
  },
  "render": [Function],
  "styledComponentId": "StyledNativeComponent",
  "target": [Function],
  "withComponent": [Function],
}
`;
